src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -704,6 +709,14 @@ public DataflowAnalyzedMethodNode DataflowAnalyzedMethod(MethodIL methodIL)
             return _dataflowAnalyzedMethods.GetOrAdd(new MethodILKey(methodIL));
         }
 
+        private NodeCache<TypeDesc, DataflowAnalyzedTypeDefinitionNode> _dataflowAnalyzedTypeDefinitions;
+
+        public DataflowAnalyzedTypeDefinitionNode DataflowAnalyzedTypeDefinition(TypeDesc type)
+        {
+            Debug.Assert(type.IsTypeDefinition);
```

Comment:
Nit: we check this in the node constructor - it's the more appropriate place to check node invariants.